tests/h/security/test_acl.py

```diff
@@ -24,82 +24,104 @@ class TestACLForUser:
     @pytest.mark.parametrize(
         "permission", (Permission.User.UPDATE, Permission.User.READ)
     )
-    def test_it(self, permits, user, principal_template, permission, is_permitted):
+    def test_it(self, user_permits, user, principal_template, permission, is_permitted):
         principal = principal_template.format(user=user)
-        assert bool(permits([principal, "some_noise"], permission)) == is_permitted
+        assert bool(user_permits([principal, "some_noise"], permission)) == is_permitted
 
     @pytest.fixture
-    def permits(self, permits, user):
-        return functools.partial(permits, ACL.for_user(user))
+    def user_permits(self, permits, user):
+        return functools.partial(permits, ObjectWithACL(ACL.for_user(user)))
 
     @pytest.fixture
     def user(self, factories):
         return factories.User.create()
 
 
 class TestACLForGroup:
-    def test_authority_joinable(self, group, permits):
+    def test_logged_in_users_get_create_permission(
+        self, group_permits, no_group_permits
+    ):
+        assert group_permits([role.User], Permission.Group.CREATE)
+        assert not group_permits([], Permission.Group.CREATE)
+        assert no_group_permits([role.User], Permission.Group.CREATE)
+        assert not no_group_permits([], Permission.Group.CREATE)
+
+    def test_logged_in_users_get_upsert_permission_when_theres_no_group(
+        self, no_group_permits
+    ):
+        assert no_group_permits([role.User], Permission.Group.UPSERT)
+        assert not no_group_permits([], Permission.Group.UPSERT)
```

Comment:
This is part of why this code is longer than before. I think this testing was missed